Added support for git clone over SSH.

jjekircp opened the pull request:
This is part of a proposed fix for ros_buildfarm [issue #91](https://github.com/ros-infrastructure/ros_buildfarm/issues/91), adding support for cloning source and release repositories over git+ssh.

This provides a credential in Jenkins with an SSH private key as well as setting up in the SSH known hosts on the slave so jobs can git clone without any user intervention.

The parameters added in this PR can be omitted entirely for users who do not need this functionality.

tfoote commented on slave/manifests/site.pp:
```diff
@@ -41,6 +41,25 @@
+  owner => 'jenkins-slave',
+  group => 'jenkins-slave',
+  content => template('slave_files/known_hosts.erb'),
+  require => User['jenkins-slave'],
```
This needs to require File['/home/jenkins-slave/.ssh/'] which in turn relies on the User['jenkins-slave'] so you can just substitute it.

tfoote commented:
Thanks this looks pretty good. One question I have is why you switched to looking up the config values in the puppet files instead of in the erb templates?

jjekircp commented:
It wasn't initially clear to me that it was possible to specify a default value for the `scope.function_hiera` call, so I switched in the course of figuring out how to make all the new config values I added to `master/common.yaml` and `slave/common.yaml` optional.

I don't have much of a background with Puppet or Ruby, so after burning a bunch of time to get that to work I left it as is. It wouldn't be hard to change back to doing the hiera lookups in the template now though, if that's preferred here.

tfoote commented:
That would be great to keep the logic in one location.

jjekircp commented:
Done. Tried eb1248e on my dev system with the parameters present and absent and I'm still getting the correct behaviour.

tfoote commented:
Great thanks. This looks good, but I likely won't have time to test it until after ROSCon.

tfoote commented on master/manifests/site.pp:
```diff
@@ -284,6 +284,10 @@
 }
 
 jenkins::plugin {
+  'workflow-step-api': ;
```
Is this necessary?

dirk-thomas replied:
Why is this plugin needed?

jjekircp replied:
At the time I added that line, the Puppet module that installed Jenkins did not install plugin dependencies automatically; they all had to be specified explicitly (sorry, I don't have a documentation reference to hand). This is a dependency of the SSH Agent Plugin per https://wiki.jenkins-ci.org/display/JENKINS/SSH+Agent+Plugin
